src/Controls/tests/TestCases/Issues/Issue21109.xaml.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using Microsoft.Maui;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Controls.Xaml;
+
+namespace Maui.Controls.Sample.Issues
+{
+	[XamlCompilation(XamlCompilationOptions.Compile)]
+	[Issue(IssueTracker.Github, 21109, "[Android] MAUI 8.0.3 -> 8.0.6 regression: custom handler with key listener no longer works", PlatformAffected.All)]
+	public partial class Issue21109 : ContentPage
+	{
+		public Issue21109()
+		{
+			InitializeComponent();
+
+			ReturnTypeResult.Text = $"ReturnType: {ReturnTypeEntry.ReturnType}";
+		}
+
+		void OnReturnTypeEntryTextChanged(object sender, TextChangedEventArgs e)
+		{
+			Random rnd = new Random();
+			var returnTypeCount = Enum.GetNames(typeof(ReturnType)).Length;
+
+			ReturnType returnType = ReturnType.Default;
+
+			do
+			{
+				returnType = (ReturnType)rnd.Next(0, returnTypeCount);
+			} while (returnType == ReturnTypeEntry.ReturnType);
+
+			ReturnTypeEntry.ReturnType = returnType;
+			ReturnTypeResult.Text = $"ReturnType: {returnType}";
+		}
+	}
+
+	public class DecimalEntry : Entry
```

Comment:
I wonder if this is better named with the issue number so it is all linked:

```suggestion
	public class Issue21109DecimalEntry : Entry
```

Reply:
Maybe also the mapper should be here in this file as well like an extension method:

```cs
public static class Issue21109Extensions
{
    public static MauiAppBuilder Issue21109AddMappers(this MauiAppBuilder builder)
    {
        builder.ConfigureMauiHandlers(handlers => {
			Microsoft.Maui.Handlers.EntryHandler.Mapper.AppendToMapping(nameof(DecimalEntry), (handler, view) =>
			{
				if (view is DecimalEntry)
				{
#if ANDROID
					handler.PlatformView.KeyListener = new Platform.NumericKeyListener(handler.PlatformView.InputType);
#endif
				}
			});
        });
    }
}
```

Reply:
Yeah, renamed the custom Entry control and moved the Mapper to the same class.